frmts/gti/gdaltileindexdataset.cpp

```diff
@@ -3864,17 +3864,18 @@ bool GDALTileIndexDataset::CollectSources(double dfXOff, double dfYOff,
                              m_aoSourceDesc.begin() + i);
     }
 
-    // Truncate the array when its last elements have no dataset
-    i = m_aoSourceDesc.size();
-    while (i > 0)
+    // Remove elements that have no dataset
+    size_t j = 0;
+    for (i = 0; i < m_aoSourceDesc.size(); ++i)
```

Comment:
Code looks good but have you considered using std::remove_if instead?

Reply:
good suggestion, done